### PR TITLE
Update transaction_retry dependencies

### PR DESCRIPTION
**Pull request by bduran82**
Adapted forked gem to OW organisation and updated gem dependencies:

* Support activerecord >= 5.1
* Support ruby 2.6.2
* Upgrade dependency version of transaction_isolation to 1.0.5 and adapt affected code
* Upgrade minitest
* Adapt gemspec info

### CODE REVIEW

**Comment by bduran82** [comment]
[(2) PDI: Avoid mysql deadlock errors](https://trello.com/c/muPoHi4q/504-2-pdi-avoid-mysql-deadlock-errors)

File: transaction_retry.gemspec [hunk 1]
```diff
@@ -5,18 +5,18 @@ require "transaction_retry/version"
...
-  s.email       = ["qertoip@gmail.com"]
-  s.homepage    = "https://github.com/qertoip/transaction_retry"
+  s.authors     = ["Optimal Workshop"]
+  s.email       = []
+  s.homepage    = "https://github.com/optimalworkshop/transaction_retry"
   s.summary     = %q{Retries database transaction on deadlock and transaction serialization errors. Supports MySQL, PostgreSQL and SQLite.}
   s.description = %q{Retries database transaction on deadlock and transaction serialization errors. Supports MySQL, PostgreSQL and SQLite (as long as you are using new drivers mysql2, pg, sqlite3).}
-  s.required_ruby_version = '>= 1.9.2'
-  
+  s.required_ruby_version = '>= 2.6.2'
```
Comment (raspygold) [inline transaction_retry.gemspec]: Current CI environment only supports `2.6.1` would it be possible to take a similar approach to AR version with `>= 2.6` instead?

File: transaction_retry.gemspec [hunk 2]
```diff
@@ -5,18 +5,18 @@ require "transaction_retry/version"
 Gem::Specification.new do |s|
   s.name        = "transaction_retry"
   s.version     = TransactionRetry::VERSION
-  s.authors     = ["Piotr 'Qertoip' Włodarek"]
-  s.email       = ["qertoip@gmail.com"]
-  s.homepage    = "https://github.com/qertoip/transaction_retry"
+  s.authors     = ["Optimal Workshop"]
```
Comment (raspygold) [inline transaction_retry.gemspec]: Should Piotr's name be left in this list and ours added instead of replacing their name?

Comment (bduran82) [reply transaction_retry.gemspec]: @raspygold I kept his name in the licence file (file that should we copy over under the MIT licence), but a part from that, this repo is already new software that OW is maintaining and not him, so we should put our organisation instead. 

I'll open an issue on the main gem's repo and suggest our changes, if they want to merge our fork then of course we should switch back the author 😬

**Comment by bduran82** [comment]
Thanks @raspygold 

I decreased the minimal ruby version to 2.2.2 (the minimal required by Rails5). Ready for you to review again.

**Comment by bduran82** [comment]
@raspygold Updated README (also added there a mention to the forked repo) and I also took the liberty to add a CHANGELOG.md (please let me know if you request any change on that regard and I'll do it in a new PR)

**Pull request merged**